migrate_nidirect_utils/src/Command/NidirectMigratePostPublishStatusCommand.php

```diff
@@ -32,54 +56,138 @@ protected function configure() {
+    $query = $this->dbConnDrupal8->update('node_field_data')
+      ->fields(['status' => $status])
+      ->condition('nid', $nid)
+      ->execute();
+
+    $query = $this->dbConnDrupal8->update('node_field_revision')
+      ->fields(['status' => $status])
+      ->condition('nid', $nid)
+      ->condition('vid', $d8_vid)
+      ->execute();
+
+    // Get the D7 revision id.
+    $vid = $this->dbConnMigrate->query(
+      "SELECT vid FROM {node} WHERE nid = :nid", [':nid' => $nid]
+    )->fetchField();
+
+    // Update the current revision if necessary.
+    if ($vid != $d8_vid) {
+      $vid = $this->updateCurrentRevision($nid, $vid, $d8_vid);
+    }
+
+    // The 'revision_translation_affected' field is poorly documented (and
+    // understood) in Drupal core, and is sometimes set to NULL after migrating
+    // content from Drupal 7. There is much discussion at
+    // https://www.drupal.org/project/drupal/issues/2746541 but after testing
+    // and investigation I am yet to find a case where it should not be
+    // set to '1'.
+    // Hence, we set it to '1' across the board to solve the problem
+    // of revisions not appearing on the revisions tab.
+    $query = $this->dbConnDrupal8->update('node_field_revision')
+      ->fields(['revision_translation_affected' => 1])
+      ->condition('nid', $nid)
+      ->execute();
+
+    // Make sure that we have a 'published' revision.
+    $query = $this->dbConnDrupal8->update('content_moderation_state_field_data')
+      ->fields(['moderation_state' => 'published'])
+      ->condition('content_entity_id', $nid)
+      ->condition('content_entity_revision_id', $vid)
+      ->execute();
+
+    $query = $this->dbConnDrupal8->update('content_moderation_state_field_revision')
+      ->fields(['moderation_state' => 'published'])
+      ->condition('content_entity_id', $nid)
+      ->condition('content_entity_revision_id', $vid)
+      ->execute();
+  }
+
+  /**
+   * Updates the current revision for the given node.
+   *
+   * @param int $nid
+   *   The node id.
+   * @param int $vid
+   *   The target revision id (from D7).
+   * @param int $d8_vid
+   *   The current D8 revision id.
+   *
+   * @return string
+   *   Current revision id.
+   */
+  private function updateCurrentRevision(int $nid, int $vid, int $d8_vid) {
+    // Does this revision exist in D8 ?
+    $check_vid = $this->dbConnDrupal8->query(
+      "SELECT vid FROM {node_field_revision} WHERE nid = :nid AND vid = :vid", [':nid' => $nid, ':vid' => $vid]
+    )->fetchField();
+    if (!empty($check_vid)) {
+      // Does the current D8 revision exist in D7 ?
+      $check_d7_vid = $this->dbConnMigrate->query(
+        "SELECT vid FROM {node_revision} WHERE nid = :nid and vid = :vid", [':nid' => $nid, ':vid' => $d8_vid]
+      )->fetchField();
+      if (!empty($check_d7_vid)) {
+        // Make the D7 revision the current revision in D8.
+        // N.B. This will only work in the 'one hit' migration scenario, it may
```

Comment:
Good point. This would be useful to share with the wider team in case there are discussions that request a rolling or phased migration. Generally speaking, once editors start working on content that has been migrated it becomes really hard to consolidate any changes so an approach of "if you edit it, you now own it" helps to manage expectations.

Reply:
Initially the plans were to have a rolling migration but it looks like the plan has changed to a complete switch over the course of a weekend.